src/metabase/models/params.clj

```diff
@@ -100,25 +100,25 @@
        field-ids->param-field-values-ignoring-current-user
        params.field-values/field-id->field-values-for-current-user) param-field-ids)))
 
-(defn- template-tag->field-form
+(defn- find-template-tag
   "Fetch the `:field` clause from `dashcard` referenced by `template-tag`.
 
-    (template-tag->field-form [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
+    (find-template-tag [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
   [[_ tag] card]
-  (get-in card [:dataset_query :native :template-tags (u/qualified-name tag) :dimension]))
+  (get-in card [:dataset_query :native :template-tags (u/qualified-name tag)]))
 
-(mu/defn param-target->field-clause :- [:maybe mbql.s/Field]
+(mu/defn param-target->template-tag :- [:maybe [:or mbql.s/TemplateTag
+                                                [:map [:dimension mbql.s/Field]]]]
   "Parse a Card parameter `target` form, which looks something like `[:dimension [:field-id 100]]`, and return the Field
   ID it references (if any)."
   [target card]
   (let [target (mbql.normalize/normalize-tokens target :ignore-path)]
     (when (mbql.u/is-clause? :dimension target)
       (let [[_ dimension] target]
         (try
-          (unwrap-field-or-expression-clause
-           (if (mbql.u/is-clause? :template-tag dimension)
-             (template-tag->field-form dimension card)
-             dimension))
+          (if (mbql.u/is-clause? :template-tag dimension)
+            (find-template-tag dimension card)
+            {:dimension    (unwrap-field-or-expression-clause dimension)})
```

Comment:
this is misleading because not all `templateTag` has a dimension, it's only applied for fieldFilter.

See this schema for all types of template tag https://github.com/metabase/metabase/blob/ngoc-optimize-revision/shared/src/metabase/mbql/schema.cljc#L1207

Reply:
Misleading as in a function name? Or as test inside of `if`?

Reply:
yea, the function name say it returns a template tag, but a map with `:dimension` key is not a template tag. it's not even satisfied as one of type of template tag.